### ISSUE TITLE
[Optimum Neuron] Compilation error when using a label smoothing factor

### ISSUE THREAD

**Issue by michaelbenayoun** [aws-neuron/aws-neuron-sdk#643]
When using `optimum-neuron`, the following command leads to a compilation error:

```bash
python examples/summarization/run_summarization.py --model_name_or_path facebook/bart-base --dataset_name cnn_dailymail --dataset_config_name 3.0.0 --do_train --per_device_train_batch_size 1 --per_device_eval_batch_size 1 --max_source_length 1024 --max_target_length 128 --overwrite_output_dir --output_dir test_bart_base --bf16 --label_smoothing_factor 0.001
```

Stack trace:
```
 To fix the error:
 1. Raise the error with the AWS Neuron team.
 2. If the error is already fixed, you can retry compilation by passing --retry_failed_compilation in NEURON_CC_FLAGS
2023-03-20 14:14:58.418509: W tensorflow/core/framework/op_kernel.cc:1830] OP_REQUIRES failed at tpu_execute_op.cc:266 : INTERNAL: neuronx-cc compilation failed.
  0%|                                                                                                                                                  | 2/861339 [00:02<270:21:19,  1.13s/it$
2023-03-20 14:14:58.551489: E tensorflow/compiler/xla/xla_client/xla_util.cc:90] StackTrace:
2023-03-20 14:14:58.551531: E tensorflow/compiler/xla/xla_client/xla_util.cc:90] *** Begin stack trace ***
2023-03-20 14:14:58.551536: E tensorflow/compiler/xla/xla_client/xla_util.cc:90]        tsl::CurrentStackTrace()
2023-03-20 14:14:58.551546: E tensorflow/compiler/xla/xla_client/xla_util.cc:90]        xla::util::ReportComputationError(tsl::Status const&, absl::lts_20220623::Span<xla::XlaComputation co$
st* const>, absl::lts_20220623::Span<xla::Shape const* const>)
2023-03-20 14:14:58.551553: E tensorflow/compiler/xla/xla_client/xla_util.cc:90]        xla::XrtComputationClient::ExecuteComputation(xla::ComputationClient::Computation const&, absl::lts_2$
220623::Span<std::shared_ptr<xla::ComputationClient::Data> const>, std::string const&, xla::ComputationClient::ExecuteComputationOptions const&)
2023-03-20 14:14:58.551562: E tensorflow/compiler/xla/xla_client/xla_util.cc:90] 
2023-03-20 14:14:58.551567: E tensorflow/compiler/xla/xla_client/xla_util.cc:90]        xla::util::MultiWait::Complete(std::function<void ()> const&)
2023-03-20 14:14:58.551573: E tensorflow/compiler/xla/xla_client/xla_util.cc:90] 
2023-03-20 14:14:58.551578: E tensorflow/compiler/xla/xla_client/xla_util.cc:90] 
2023-03-20 14:14:58.551585: E tensorflow/compiler/xla/xla_client/xla_util.cc:90] 
2023-03-20 14:14:58.551593: E tensorflow/compiler/xla/xla_client/xla_util.cc:90]        clone
2023-03-20 14:14:58.551608: E tensorflow/compiler/xla/xla_client/xla_util.cc:90] *** End stack trace ***
2023-03-20 14:14:58.551613: E tensorflow/compiler/xla/xla_client/xla_util.cc:90] 
2023-03-20 14:14:58.551622: E tensorflow/compiler/xla/xla_client/xla_util.cc:90] Status: INTERNAL: From /job:localservice/replica:0/task:0:
2023-03-20 14:14:58.551627: E tensorflow/compiler/xla/xla_client/xla_util.cc:90] 2 root error(s) found.
2023-03-20 14:14:58.551632: E tensorflow/compiler/xla/xla_client/xla_util.cc:90]   (0) INTERNAL: neuronx-cc compilation failed.
2023-03-20 14:14:58.551637: E tensorflow/compiler/xla/xla_client/xla_util.cc:90]         [[{{node XRTExecute}}]]
2023-03-20 14:14:58.551643: E tensorflow/compiler/xla/xla_client/xla_util.cc:90]         [[XRTExecute_G15]]
2023-03-20 14:14:58.551648: E tensorflow/compiler/xla/xla_client/xla_util.cc:90]   (1) INTERNAL: neuronx-cc compilation failed.
2023-03-20 14:14:58.551656: E tensorflow/compiler/xla/xla_client/xla_util.cc:90]         [[{{node XRTExecute}}]]
2023-03-20 14:14:58.551660: E tensorflow/compiler/xla/xla_client/xla_util.cc:90] 0 successful operations.
2023-03-20 14:14:58.551669: E tensorflow/compiler/xla/xla_client/xla_util.cc:90] 0 derived errors ignored.
2023-03-20 14:14:58.551673: E tensorflow/compiler/xla/xla_client/xla_util.cc:90] Recent warning and error logs:
2023-03-20 14:14:58.551682: E tensorflow/compiler/xla/xla_client/xla_util.cc:90]   OP_REQUIRES failed at tpu_execute_op.cc:266 : INTERNAL: neuronx-cc compilation failed.

```

Current observations:

- It fails at the second step of training 
- The same command line works on GPU so it seems to be neuron-dependent
- When removing the `--label_smoothing_factor 0.001` argument, it works

More on this issue here => https://github.com/huggingface/optimum-neuron/issues/19

**Comment by mrnikwaws** [comment]
@michaelbenayoun we are continuing to work on this issue - no actionable updates yet, but I can confirm I can reproduce the issue.

**Comment by aws-taylor** [comment]
Hello @michaelbenayoun,

 We believe we have fixed the issue and it is no longer failing in our internal tests. We'll update this issue when the fix is available in a future release.
 
 -T

**Comment by michaelbenayoun** [comment]
Thanks @aws-taylor

**Comment by awsilya** [comment]
Hi @michaelbenayoun.  The fix should be available in just released Neuron 2.10.  Please try it.

**Comment by aws-mvaria** [comment]
Closing this issue. Please re-open if you encounter any further issues. Thanks.
